File: January 2021/31.01.2021/Kaan.cpp
class Solution {
public:
    int countBattleships(vector<vector<char>>& board) {
        int result = 0;    
        for(int i = 0 ;i<board.size();i++){       
            for(int j = 0 ; j<board[0].size();j++){
                if(board[i][j] =='X'){
                    if(j==0 && i == 0){
                        result+=1;
                    }
                    else if(j==0){
                        if(board[i-1][j] != 'X'){
                            result+=1;
                        }
                    }
                    else if(i==0){
                        if(board[i][j-1] != 'X'){
                            result+=1;
                        }
                    }
                    else{
                        if(board[i-1][j] != 'X' && board[i][j-1] != 'X'){
                            result+=1;
                        }
                    } 
                }
            }  
        }
        return result;     
    }
};

File: March 2021/06.03.2021/Kaan.java
class Solution {
    public int minimumLengthEncoding(String[] words) {
        int result = 0;
        HashSet<String> wordSet = new HashSet<>();
        for(int i=0 ; i<words.length;i++){
            boolean found = false;
            for(int j = 0 ; j<words.length;j++){
                if(i == j){
                    continue;
                }
                else if(words[i].equals(words[j])){
                    continue;
                }
                else if(words[j].endsWith(words[i])){
                    found = true;
                    break;
                }
            }
            if(!found && !wordSet.contains(words[i])){
                wordSet.add(words[i]);
                result+=words[i].length() +1;
            }
        }
        return result; 
    }
}
